frontend/src/components/visual-editor/v2/Diagrams.tsx

```diff
@@ -80,6 +83,27 @@ const Diagrams = () => {
   const { searchPayload } = useSearch<IBlock>({
     $eq: [{ category: selectedCategoryId }],
   });
+  const selectedEntities = engine?.getModel()?.getSelectedEntities();
+  const selectedLinks = (selectedEntities || []).filter(
+    (entity) => entity instanceof AdvancedLinkModel,
+  );
+  const selectedBlocks = (selectedEntities || []).filter(
+    (entity) => entity instanceof NodeModel,
+  );
+  const { toast } = useToast();
+  const { mutate: duplicateBlock, isLoading: isDuplicatingBlock } = useCreate(
+    EntityType.BLOCK,
+    {
+      onError: () => {
+        toast.error(t("message.duplicate_block_error"));
+      },
+    },
+  );
+  const shouldDisableDuplicateButton =
+    selectedLinks.length >= 1 ||
+    selectedBlocks.length > 1 ||
+    selectedBlocks.length === 0 ||
+    isDuplicatingBlock;
```

Comment:
```suggestion
  const selectedEntities = getSelectedIds();
  const shouldDisableDuplicateButton =
    selectedEntities.length !== 1 ||
    selectedEntities[0]?.length !== 24 ||
    isDuplicatingBlock;
```

Need to be placed after the JSX